### ISSUE TITLE
I implemented an option to paint the title bar with the window color

### ISSUE THREAD

**Issue by Petross404** [ishovkun/SierraBreeze#40]
I like very much when my decoration seems to "fit" visually with my window's colors. So I took the liberty to create an option where the user can choose to keep the default style or to match the colors. 

It looks like this : 

![Alt text](https://i.imgur.com/iF9BLGr.png)

and this : 

![Alt text](https://i.imgur.com/JcDlZgu.png)

It's a click away to revert 

![Alt text](https://i.imgur.com/WRCMGsh.png)

Ofcourse the konsole is respected (nice work btw with that!)

![Alt text](https://i.imgur.com/vusFIuQ.png)

I will open a PR for this. Your c&c are more than welcome.

**Comment by Petross404** [comment]
Ok, one thing I noticed. It doesn't respect the color of the window if the user changes it inside the application - e.g. Krita has this functionality. It's titlebar's color stays the same though. 

Any ideas how to force a repaint event?

**Comment by Petross404** [comment]
Ok, I made `painter` and `repaintRegion` private members of the class. This code snippet inside `Decoration::init`
```
connect(c, static_cast<void (KDecoration2::DecoratedClient::*)(const QPalette&)>(&KDecoration2::DecoratedClient::paletteChanged), this,
                [this]
        {
            this->paintTitleBar(&painter, repaintRegion);
        });
```
helped but for KDevelop, Krita still refuses to cooperate.

**Comment by trmdi** [comment]
The question is: what is wrong with changing the titlebar color from the KDE Color setting? It's much more customizable than this, users can choose any color they want.

**Comment by Petross404** [comment]
@ishovkun I am thinking if with kf5::plasma I can read the background colors of non qt applications as well. Also instead of qpallete signal I will try with `void 	colorsChanged() `. For more look here https://api.kde.org/frameworks/plasma-framework/html/classColorScope.html

**Comment by yaabdalla** [comment]
Sorry to thread steal, but I can't seem to get SierraBreeze to match even Konsole's colors for the title bar. Any help? Also doesn't match for any darker themed application (IDEA).

![screenshot_20181227_035224](https://user-images.githubusercontent.com/6026752/50475615-032e9780-098b-11e9-9214-a25ec1158782.png)
![screenshot_20181227_035258](https://user-images.githubusercontent.com/6026752/50475626-075ab500-098b-11e9-9b7f-f22e6ba5e333.png)

**Comment by treenix** [comment]
@Petross404 This error occurs to me as well. I just installed manjaro KDE and sierrabreeze, nothing else changed. I only get it working using this [workaround](https://github.com/kupiqu/SierraBreezeEnhanced/issues/5#issuecomment-445437687).

**Comment by Petross404** [comment]
@yaabdalla IDEA is a java IDE? This patch works only with Qt applications. Other than that, konsole title bar should work as it's shown in my post.

**Comment by gcjuan** [comment]
In my case the terminal looks pretty awesome **with SierraBreeze** as windows decoration:

![imagen](https://user-images.githubusercontent.com/19599325/52866957-14e91b80-3140-11e9-9939-5e090465fc1a.png)

It has rounded corners everywhere and the title bar adopts the color of the window's content.

It only happens with the terminal. For other apps is not the case (although I have checked the option "Match Title Bar's and Window's color" on the SierraBreeze options menu):

![imagen](https://user-images.githubusercontent.com/19599325/52867481-53330a80-3141-11e9-9e70-67498b0f340b.png)

You can see that the title bar is not adjusting its color to the same as the window has as it occurs with the terminal window. It also has a black border on the bottom which obviously has the same color as the title bar. This is not desirable, but I don't wanted to set the borders to "no borders" cause if I do it so, the rounded bottom borders disappear and they give a nice "touch" to the whole.

By using [Kvantum](https://github.com/tsujan/Kvantum) (A Linux SVG-based theme engine for Qt4/Qt5 and KDE), the Kvantum theme [KvGlass](https://store.kde.org/p/1201321/) and its windows decoration theme [Aurorae Glass-Kv](https://store.kde.org/p/1201322/) ([modified by my with OSX buttons](https://github.com/ishovkun/SierraBreeze/files/2869875/Glass-Kv-OSX-mod.zip)) the look&feel is as I would expect from the SierraBreeze's option mentioned above ("Match Title Bar's and Window's color"):

![imagen](https://user-images.githubusercontent.com/19599325/52867822-0bf94980-3142-11e9-9673-59a3040815f4.png)

The problem with the windows decoration theme I modified is, that it has a fixed "blurred" color which in fact is the same blurred color set by Kvantum's theme (that's why it looks the same applied to Dolphin, both window and titlebar have the same background color). Here a clear example of inconsistency caused of non-adapting Aurorae themes, terminal's background differs from titlebar's one in this case. They are close to each other but not exactly the same:

![imagen](https://user-images.githubusercontent.com/19599325/52869202-6cd65100-3145-11e9-99d0-3b5f34775e72.png)

It would be perfect if SierraBreeze could achieve the same results as the mentioned "Kvantum-mix" does.

Why is SierraBreeze adapting the titlebar's color (and thus bottom borders accordingly) on the first terminal screenshot but not working on other cases? Could anyone help me here or tell me if there is something that we can modify on SierraBreeze's code in order to get these results?

Thanks in advance =)

**Comment by Petross404** [comment]
Is [this](https://user-images.githubusercontent.com/19599325/52867481-53330a80-3141-11e9-9e70-67498b0f340b.png) using kvantum or breeze? Correct me if I am wrong but SierraBreeze (and this option) works with the breeze theme.

The titlebar [get's](https://github.com/ishovkun/SierraBreeze/blob/master/breezedecoration.cpp#L580) it's color from the window's background color. As you can see in the code, the titlebar doesn't know anything about the window's _opacity_ or _blur_. 

There is a project on github called [BreezeBlurred](https://github.com/alex47/BreezeBlurred) which can provide blurred titlebars with breeze theme, but AFAIK it doesn't help you by matching titlebar's color. Sure you can append a color per application basis but that's what I wanted to avoid and I implemented this option. Maybe Alex can help with this if you choose to go for his solution.

There is a chance that I missed something, so let me know if I haven't actually answered you. Have a nice day.

**Comment by gcjuan** [comment]
Hi! Thanks for you response.

In both the 1st and 2nd screenshots of my post I'm using Kvantum as theme (as widget style) and SierraBreeze.

My doubt was rather, how is it possible, that in the first screenshot the title bar has adopted the color of the terminal but not in the second one with Dolphin? Which are the criteria of matching a window's color? What does exactly mean "palette().color" that you reference me in the code and which color is taken as window's reference color?

Thanks a lot and I hope not to annoy you ;)

**Comment by Petross404** [comment]
No, you don't annoy me. It's just that I may disappoint a user because I am not **the** best programmer. Anyway, I will try to help you.

Below `auto c` represents the window. After that, I construct the titlebar's color based on `c`'s `QPalette::Window` which is a way to read the (Qt) window's color in computerish.
```cpp
        const auto c = client().data();
        const QColor matchedTitleBarColor(c->palette().color(QPalette::Window));
```
So, there is that.

> Which are the criteria of matching a window's color?

As long as it's a Qt app.

> how is it possible, that in the first screenshot the title bar has adopted the color of the terminal but not in the second one with Dolphin?

I guess it's because [here](https://github.com/ishovkun/SierraBreeze/blob/master/breezedecoration.cpp#L305) SierraBreeze reads the konsole config file, in order to find the correct colors to paint the titlebar. What confuses me is that it doesn't seem to do something about blurring. I don't know what to advise you. Maybe ask help from the original author and I will happily participate to help you with his guidance.
